core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregatorBuilder.java

```diff
@@ -57,6 +58,8 @@ public FiltersAggregatorBuilder(String name, KeyedFilter... filters) {
 
     private FiltersAggregatorBuilder(String name, List<KeyedFilter> filters) {
         super(name, InternalFilters.TYPE);
+        // internally we want to have a fixed order of filters, regardless of the order of the filters in the request
+        Collections.sort(filters);
```

Comment:
sure, will remove the Comparable. Copying is not necessary though in this case, this is a privat ctor, and the other ctor calling it already converts the input array into a list.